tensorbay/opendataset/VGGFace2/loader.py

```diff
@@ -0,0 +1,232 @@
+    root_path: str,
+) -> Tuple[Dict[str, str], Dict[str, Dict[str, object]]]:
+    class_id_name = {}
+    class_name_attrs = {}
+    with open(os.path.join(root_path, "label", "identity_meta.csv"), "r", encoding="utf-8") as fp:
+        lines = csv.reader(fp)
+        next(lines)
+        for meta_identities in lines:
+            if meta_identities[-1] != "":
+                # The normal format of each line of the file is
+                # '<class_id>,"<class_name>",<sample_num>,<flag>,<gender>,\n'
+                # but now there is an error type
+                # '<class_id>,"<class_,name>",<sample_num>,<flag>,<gender>\n'
+                # join the splitted "class_ and name"
+                meta_identities[1] = "".join(meta_identities[1:3])
+                # Starting from the 4th element, each element moves forward one step
+                meta_identities.pop(2)
+                meta_identities[4] = meta_identities[4].rstrip("\n")
+            class_name = meta_identities[1].strip('"')
+            class_id_name[meta_identities[0]] = class_name
+            class_name_attrs[class_name] = dict(
+                zip(
+                    CLASS_ATTRS,
+                    (
+                        int(meta_identities[2]),
+                        bool(int(meta_identities[3])),
+                        meta_identities[4],
+                    ),
+                )
+            )
+    return class_id_name, class_name_attrs
+
+
+def _get_box2d_cords(root_path: str) -> Dict[str, List[int]]:
+    box2d_file_cords = {}
+    for file in BOX2D_FILES:
+        with open(os.path.join(root_path, file), "r", encoding="utf-8") as fp:
+            lines = csv.reader(fp)
+            next(lines)
+            for box2d_cords in lines:
+                box2d_file_cords[f"{box2d_cords[0]}.jpg"] = [int(x) for x in box2d_cords[1:]]
+    return box2d_file_cords
+
+
+def _get_box2d_attrs(root_path: str) -> DefaultDict[str, _NEW]:
+    box2d_file_attrs: DefaultDict[str, _NEW] = defaultdict(dict)
+    box2d_files2 = glob(os.path.join(root_path, "label", "attributes", "*.txt"))
+    for file in box2d_files2:
+        attr = os.path.basename(file).rstrip(".txt")
+        with open(file, "r", encoding="utf-8") as fp:
+            lines = fp.readlines()
+            for line in lines:
+                box_attrs = line.rstrip("\n").split("\t")
+                box2d_file_attrs[BOX2D_ATTRS[0]][attr] = bool(int(box_attrs[1]))
+    return box2d_file_attrs
+
+
+def _get_keypoints2d_keypoints(root_path: str) -> Dict[str, List[Tuple[int, ...]]]:
+    ketpoints2d_file_keypoints = {}
+    for file in KEYPOINTS2D_FILES:
+        with open(os.path.join(root_path, file), "r", encoding="utf-8") as fp:
+            lines = csv.reader(fp)
+            next(lines)
+            for key_points2d in lines:
+                ketpoints2d_file_keypoints[f"{int(key_points2d[0])}.jpg"] = list(
+                    chunked([int(x) for x in key_points2d[1:]], 2)
```

Comment:
use generator expression rather than list comprehension for saving memory.
for the same reason, use `islice` rather `slice`